crates/biome_migrate/src/analyzers/nursery_rules.rs

```diff
@@ -0,0 +1,299 @@
+use crate::{declare_migration, MigrationAction};
+use biome_analyze::context::RuleContext;
+use biome_analyze::{ActionCategory, Ast, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_diagnostics::{category, Applicability};
+use biome_json_factory::make::{
+    json_member, json_member_list, json_member_name, json_object_value, json_string_literal, token,
+};
+use biome_json_syntax::{AnyJsonValue, JsonMember, JsonMemberList, JsonRoot, T};
+use biome_rowan::{
+    AstNode, AstNodeExt, AstSeparatedList, BatchMutationExt, TextRange, TriviaPieceKind, WalkEvent,
+};
+use std::collections::HashMap;
+
+#[cfg(not(debug_assertions))]
+use crate::analyzers::rules_to_migrate::RULES_TO_MIGRATE;
+
+declare_migration! {
+    pub(crate) NurseryRules {
+        version: "1.7.0",
+        name: "nurseryRules",
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct MigrateRuleState {
+    /// The member of the new rule
+    nursery_rule_member: JsonMember,
+    /// The member of the group where the new rule should be moved
+    nursery_group: JsonMember,
```

Comment:
I am a bit worried about the fact that we have tightened the implementation to JSON config. For the time being it is ok and I don't see how we could solve it anyway.

Reply:
I really don't understand what you mean 😅

Reply:
I mean we operate over JSON nodes, if one day we also handle YAML or TOML configurations, we will need to write a rule for every format.